MapboxDirections/MBDirections.swift

```diff
@@ -192,32 +162,18 @@ open class Directions: NSObject {
      - returns: The data task for the URL.
      - postcondition: The caller must resume the returned task.
      */
-    fileprivate func dataTask(with url: URL, completionHandler: @escaping (_ json: JSONDictionary) -> Void, errorHandler: @escaping (_ error: NSError) -> Void) -> URLSessionDataTask {
+    fileprivate func dataTask(with url: URL, completionHandler: @escaping (_ data: Data) -> Void, errorHandler: @escaping (_ error: NSError) -> Void) -> URLSessionDataTask {
         
         var request = URLRequest(url: url)
         request.setValue(userAgent, forHTTPHeaderField: "User-Agent")
         return URLSession.shared.dataTask(with: request as URLRequest) { (data, response, error) in
-            var json: JSONDictionary = [:]
-            if let data = data, response?.mimeType == "application/json" {
-                do {
-                    json = try JSONSerialization.jsonObject(with: data, options: []) as! JSONDictionary
-                } catch {
-                    assert(false, "Invalid data")
-                }
-            }
-            
-            let apiStatusCode = json["code"] as? String
-            let apiMessage = json["message"] as? String
-            guard data != nil && error == nil && ((apiStatusCode == nil && apiMessage == nil) || apiStatusCode == "Ok") else {
-                let apiError = Directions.informativeError(describing: json, response: response, underlyingError: error as NSError?)
-                DispatchQueue.main.async {
-                    errorHandler(apiError)
-                }
+            guard let data = data, response?.mimeType == "application/json" else {
+                errorHandler(Directions.informativeError(describing: [:], response: response, underlyingError: error as NSError?))
```

Comment:
This call used to take a JSON object whose `code` and `message` keys determined the message. Does that no longer matter?

Reply:
Good catch. Errors are handled more gracefully in https://github.com/mapbox/MapboxDirections.swift/pull/221/commits/1a161d6fc7694f2e2dc7597baaae6e3cdc9da34e